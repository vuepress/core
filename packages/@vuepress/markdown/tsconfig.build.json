{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "module": "CommonJS",
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["./src"],
  "references": [{ "path": "../utils/tsconfig.build.json" }]
}
