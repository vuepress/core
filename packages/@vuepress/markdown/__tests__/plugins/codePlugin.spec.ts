import * as MarkdownIt from 'markdown-it'
import { codePlugin } from '@vuepress/markdown'

const codeFence = '```'

describe('@vuepress/markdown > plugins > codePlugin', () => {
  describe('plugin options', () => {
    const source = `\
${codeFence}
Raw text
${codeFence}

${codeFence}ts
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2-4,5-5}
const foo = 'foo'

function bar () {
  return 1024
}

const baz = () => {
  return 2048
}
${codeFence}
`

    it('should process code fences with default options', () => {
      const md = MarkdownIt().use(codePlugin)

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should disable `highlightLines`', () => {
      const md = MarkdownIt().use(codePlugin, {
        highlightLines: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should enable `lineNumbers`', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: true,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should disable `lineNumbers`', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should accpet numbers as `lineNumbers`', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: 4,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should disable `preWrapper`', () => {
      const md = MarkdownIt().use(codePlugin, {
        preWrapper: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should disable `vPre`', () => {
      const md = MarkdownIt().use(codePlugin, {
        vPre: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should always disable `highlightLines` if `preWrapper` is disabled', () => {
      const mdWithHighlightLines = MarkdownIt().use(codePlugin, {
        highlightLines: true,
        preWrapper: false,
      })
      const mdWithoutHighlightLine = MarkdownIt().use(codePlugin, {
        highlightLines: false,
        preWrapper: false,
      })

      expect(mdWithHighlightLines.render(source)).toBe(
        mdWithoutHighlightLine.render(source)
      )
    })

    it('should always disable `lineNumbers` if `preWrapper` is disabled', () => {
      const mdWithLineNumbers = MarkdownIt().use(codePlugin, {
        lineNumbers: true,
        preWrapper: false,
      })
      const mdWithoutLineNumbers = MarkdownIt().use(codePlugin, {
        lineNumbers: false,
        preWrapper: false,
      })

      expect(mdWithLineNumbers.render(source)).toBe(
        mdWithoutLineNumbers.render(source)
      )
    })
  })

  describe(':line-numbers / :no-line-numbers', () => {
    const source = `\
${codeFence}
Raw text
${codeFence}

${codeFence}:line-numbers
Raw text
${codeFence}

${codeFence}:no-line-numbers
Raw text
${codeFence}

${codeFence}ts
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts:line-numbers
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts:no-line-numbers
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}:line-numbers
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}:no-line-numbers
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}
`

    it('should work properly if `lineNumbers` is enabled by default', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: true,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should work properly if `lineNumbers` is disabled by default', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should work properly if `lineNumbers` is set to a threshold', () => {
      const md = MarkdownIt().use(codePlugin, {
        lineNumbers: 4,
      })

      expect(md.render(source)).toMatchSnapshot()
    })
  })

  describe(':v-pre / :no-v-pre', () => {
    const source = `\
${codeFence}
Raw text
${codeFence}

${codeFence}:v-pre
Raw text
${codeFence}

${codeFence}:no-v-pre
Raw text
${codeFence}

${codeFence}ts
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts:v-pre
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts:no-v-pre
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}:v-pre
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts{1,2}:no-v-pre
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}
`

    it('should work properly if `vPre` is enabled by default', () => {
      const md = MarkdownIt().use(codePlugin, {
        vPre: true,
      })

      expect(md.render(source)).toMatchSnapshot()
    })

    it('should work properly if `vPre` is disabled by default', () => {
      const md = MarkdownIt().use(codePlugin, {
        vPre: false,
      })

      expect(md.render(source)).toMatchSnapshot()
    })
  })

  describe('syntax highlighting', () => {
    const source = `\
${codeFence}
Raw text
${codeFence}

${codeFence}js
const foo = 'foo'

function bar () {
  return 1024
}
${codeFence}

${codeFence}ts
const foo: string = 'foo'

function bar (): number {
  return 1024
}
${codeFence}
`

    it('should work if highlighted code is wrapped with `<pre>`', () => {
      const highlight = jest.fn(
        (code, lang) =>
          `<pre><code>highlighted code: ${code}, lang: ${lang}</code></pre>`
      )
      const md = MarkdownIt({ highlight }).use(codePlugin)

      expect(md.render(source)).toMatchSnapshot()
      expect(highlight).toHaveBeenCalledTimes(3)
    })

    it('should work if highlighted code is not wrapped with `<pre>`', () => {
      const highlight = jest.fn(
        (code, lang) => `highlighted code: ${code}, lang: ${lang}`
      )
      const md = MarkdownIt({ highlight }).use(codePlugin)

      expect(md.render(source)).toMatchSnapshot()
      expect(highlight).toHaveBeenCalledTimes(3)
    })
  })
})
