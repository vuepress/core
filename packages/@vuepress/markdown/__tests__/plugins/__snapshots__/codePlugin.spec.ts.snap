// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@vuepress/markdown > plugins > codePlugin :line-numbers / :no-line-numbers should work properly if \`lineNumbers\` is disabled by default 1`] = `
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin :line-numbers / :no-line-numbers should work properly if \`lineNumbers\` is enabled by default 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin :line-numbers / :no-line-numbers should work properly if \`lineNumbers\` is set to a number by default 1`] = `
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin :v-pre / :no-v-pre should work properly if \`vPre\` is disabled by default 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text line-numbers-mode"><pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin :v-pre / :no-v-pre should work properly if \`vPre\` is enabled by default 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-text ext-text line-numbers-mode"><pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should disable \`highlightLines\` 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should disable \`lineNumbers\` 1`] = `
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
</div>
<div class="language-typescript ext-ts"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br><br>
  </div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should disable \`preWrapper\` 1`] = `
<pre v-pre class="language-text"><code>Raw text
</code></pre><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should disable \`vPre\` 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should enable \`lineNumbers\` according to number of code lines 1`] = `
<div class="language-text ext-text"><pre v-pre class="language-text"><code>Raw text
</code></pre>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin plugin options should process code fences with default options 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>Raw text
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>const foo = 'foo'

function bar () {
  return 1024
}

const baz = () =&gt; {
  return 2048
}
</code></pre>
  <div class="highlight-lines">
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div>
    <div class="highlight-line">&nbsp;</div><br><br><br><br>
  </div>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin syntax highlighting should work if highlighted code is not wrapped with \`<pre>\` 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre class="language-text"><code>highlighted code: Raw text
, lang: text</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-javascript ext-js line-numbers-mode"><pre v-pre class="language-javascript"><code>highlighted code: const foo = 'foo'

function bar () {
  return 1024
}
, lang: javascript</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre class="language-typescript"><code>highlighted code: const foo: string = 'foo'

function bar (): number {
  return 1024
}
, lang: typescript</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
`;

exports[`@vuepress/markdown > plugins > codePlugin syntax highlighting should work if highlighted code is wrapped with \`<pre>\` 1`] = `
<div class="language-text ext-text line-numbers-mode"><pre v-pre><code>highlighted code: Raw text
, lang: text</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br></div>
</div>
<div class="language-javascript ext-js line-numbers-mode"><pre v-pre><code>highlighted code: const foo = 'foo'

function bar () {
  return 1024
}
, lang: javascript</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
<div class="language-typescript ext-ts line-numbers-mode"><pre v-pre><code>highlighted code: const foo: string = 'foo'

function bar (): number {
  return 1024
}
, lang: typescript</code></pre>
  <div class="line-numbers"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div>
</div>
`;
